bin/picca_fast_metal_dmat.py

```diff
@@ -0,0 +1,570 @@
+#!/usr/bin/env python
+"""Compute the auto and cross-correlation of delta fields for a list of IGM
+absorption.
+
+This module follow the procedure described in sections 4.3 of du Mas des
+Bourboux et al. 2020 (In prep) to compute the distortion matrix
+"""
+import sys,os
+import time
+import argparse
+import numpy as np
+import fitsio
+
+from picca import constants, cf, utils, io
+from picca.utils import userprint
+
+
+def read_stack_deltas_table(filename) :
+    return fitsio.read(filename,"STACK_DELTAS")
+
+def calc_fast_metal_dmat(in_lambda_abs_1, in_lambda_abs_2,
+                         out_lambda_abs_1, out_lambda_abs_2,
+                         stack_table_1, stack_table_2, rebin_factor=None):
+    """Computes the metal distortion matrix.
+
+    Args:
+        in_lambda_abs_1 : str
+            Name of absorption in picca.constants in forest pixels from stack 1 (input, i.e. 'true' absorber)
+        in_lambda_abs_2 : str
+            Name of absorption in picca.constants in forest pixels from stack 2 (input, i.e. 'true' absorber)
+        out_lambda_abs_1 : str
+            Name of absorption in picca.constants in forest pixels from stack 1 (output, i.e. 'assumed' absorber, usually LYA)
+        out_lambda_abs_2 : str
+            Name of absorption in picca.constants in forest pixels from stack 2 (output, i.e. 'assumed' absorber, usually LYA)
+        stack_table_1: table
+            table with cols LOGLAM and WEIGHT for first series of deltas
+        stack_table_2: table
+            table with cols LOGLAM and WEIGHT for second series of deltas
+    Optionnal : rebin_factor
+            rebin loglam and weights
+    Returns:
+        The distortion matrix data
+        Note the global picca.cf contains the cosmology and the rp grid
+    """
+
+    loglam1=stack_table_1["LOGLAM"]
+    weight1=stack_table_1["WEIGHT"]
+    loglam2=stack_table_2["LOGLAM"]
+    weight2=stack_table_2["WEIGHT"]
+    if rebin_factor is not None :
+        n1=loglam1.size
+        loglam1 = loglam1[:(n1//rebin_factor)*rebin_factor].reshape((n1//rebin_factor),rebin_factor).mean(-1)
+        weight1 = weight1[:(n1//rebin_factor)*rebin_factor].reshape((n1//rebin_factor),rebin_factor).mean(-1)
+        n2=loglam2.size
+        loglam2 = loglam2[:(n2//rebin_factor)*rebin_factor].reshape((n2//rebin_factor),rebin_factor).mean(-1)
+        weight2 = weight2[:(n2//rebin_factor)*rebin_factor].reshape((n2//rebin_factor),rebin_factor).mean(-1)
+
+    # input
+    iz1 = (10**loglam1)/constants.ABSORBER_IGM[in_lambda_abs_1] - 1.
```

Comment:
I would rename this variable as `input_z1` (similarly for the rest of the variables). I think that name is more clear. Note that I originally thought these were some indexes.

Reply:
done